docs/source/java/memory.rst

```diff
@@ -0,0 +1,208 @@
+When enabled, a log will be kept of allocations.
+
+Arrow modules define simple logging facade for java SLF4J, configure it properly to see your logs (e.g. Logback/Log4J).
+
+Consider the following example to see how debug enabled help us with the tracking of allocators:
+
+.. code-block:: Java
+
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.IntVector;
+
+    try (BufferAllocator bufferAllocator = new RootAllocator(Integer.MAX_VALUE)) {
+        final int QUANTITY = 5;
+        try (IntVector intVector = new IntVector("int-01", bufferAllocator)) {
+            intVector.allocateNew(QUANTITY);
+            for (int i = 0; i < QUANTITY; i++) {
+                intVector.set(i, i);
+            }
+            intVector.setValueCount(QUANTITY);
+        }
+        // Fix the next code!, it is only to see the track of allocators when debug is enabled
+        IntVector intVectorV = new IntVector("int-02", bufferAllocator);
+        intVectorV.allocateNew(QUANTITY);
+        for (int i = 0; i < QUANTITY; i++) {
+            intVectorV.set(i, i);
+        }
+        intVectorV.setValueCount(QUANTITY);
+
+        BufferAllocator childAllocator = bufferAllocator.newChildAllocator("child-isolated", 0,
+                Integer.MAX_VALUE / 4);
+        IntVector intVectorV2 = new IntVector("int-isolated-01", childAllocator);
+        intVectorV2.allocateNew(QUANTITY);
+        for (int i = 0; i < QUANTITY; i++) {
+            intVectorV2.set(i, i);
+        }
+    }
+
+.. code-block::
+
+    15:49:32,755 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback-test.xml] at [file:/Users/java/source/demo/target/classes/logback-test.xml]
+    15:49:32,924 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.arrow] to DEBUG
+    11:56:48.944 [main] INFO  o.apache.arrow.memory.BaseAllocator - Debug mode enabled.
+    Exception in thread "main" java.lang.IllegalStateException: Allocator[ROOT] closed with outstanding child allocators.
+    Allocator(ROOT) 0/64/64/2147483647 (res/actual/peak/limit)
+      child allocators: 1
+        Allocator(child-isolated) 0/32/32/536870911 (res/actual/peak/limit)
+          child allocators: 0
+          ledgers: 1
+            ledger[3] allocator: child-isolated), isOwning: , size: , references: 2, life: 246918908438818..0, allocatorManager: [, life: ] holds 3 buffers.
+                ArrowBuf[10], address:140408097079352, length:8
+                ArrowBuf[8], address:140408097079328, length:32
+                ArrowBuf[9], address:140408097079328, length:24
+          reservations: 0
+      ledgers: 1
+        ledger[2] allocator: ROOT), isOwning: , size: , references: 2, life: 246tors can be named; this makes it easier to tell where an Arro918906331643..0, allocatorManager: [, life: ] holds 3 buffers.
```

Comment:
Seems something got copy-pasted into the middle here?

Reply:
Deleted